src/test/mon/MonMap.cc

```diff
@@ -237,9 +237,5 @@ TEST(MonMapBuildInitial, build_initial_mon_host_from_dns_fail) {
   cct->_conf.set_val("mon_host", "ceph.noname");
   MonMap monmap;
   int r = monmap.build_initial(cct.get(), false, std::cerr);
-#if defined(__FreeBSD__)
```

Comment:
why?

Reply:
@tchaikov 
I guess that changes to the socket code now result in returning `EINVAL` 
In the past ENOENT was returned which is the FreeBSD error code if DNS lookup does not work.

And that is then probably because somewhere in the code that error code is not passed on from the systemcall, but somewhere in between is rewritten in extra evaluation.

I have very limitted time to keep a very closed eye on all the errors in my Jenkins builds, due to regular work.
So I have no idea where this error has been introduced.

Reply:
@wjwithagen ack. could you put your analysis in the commit message?